docs/configuration/http_filters/fault_filter.rst

```diff
@@ -68,6 +68,8 @@ upstream_cluster:
   field is not in the config).
   TODO: allow runtime configuration on per entry basis for headers match.
```

Comment:
At this point I think we have too much duplication here with headers. Can we please just do the right thing here and make this work on the headers directly. In thinking about it, I don't think it's that hard. Basically the headers can optionally have both a header value key override, and a % match override. This will be useful also for the general routing code. I think this will satisfy @rshriram concerns also.